pyht/async_client.py

```diff
@@ -41,6 +42,8 @@ class AdvancedOptions:
         grpc_addr: str | None = None
         insecure: bool = False
         auto_refresh_lease: bool = True
+        on_prem_endpoint: str | None = None
+        on_prem_fallback: bool = True
```

Comment:
I don't think you need both of those.  As a matter of fact, this would be cleaner if you had only 2 members:
`grpc_addr` and `grpc_fallback_addr`.
This makes it more generic and reusable in other situations.

Reply:
Fixed to only have grpc_addr : str | None = None, and fallback_enabled: bool = False.  See above.